core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -786,7 +997,7 @@ public void setTaskDomains(List<Task> tasks, Workflow wf) {
                 String[] domains = domainstr.split(",");
                 tasks.forEach(task -> {
                     // Filter out SystemTask
-                    if (!WorkflowTask.Type.isSystemTask(task.getTaskType())) {
+                    if (!TaskType.isSystemTask(task.getTaskType())) {
```

Comment:
`task.getTaskType().isSystemTask()`

Reply:
The code is doing the same as before. `getTaskType()` is a String. Does not contain a method called `isSystemTask`. There is a possibility of refactoring the code in order to be a taskType but the change is not trivial: I believe SystemTaskType and TaskType overlap on a couple of scenarios where this String -> TaskType conversion fails. 
Happy to refactor the code to use TaskType in the future but I feel it is out of the scope of this PR. Do you think it is reasonable?

Reply:
Got it, let's not worry about in this PR, we will refactor it later.